### PR TITLE
support version manager in spb for js-sdk deploys

### PR DESCRIPTION
**Pull request by wsbrunson**
passes in an instance of `sdkVersionManager` to `smart/buttons` middleware (this all happens in smartcomponentnodeweb). 

the getButtonsMiddleware function will now use `sdkVersionManager.getOrInstallSDK` to get the latest version of the js-sdk

smartcomponentnodeweb is using the remote config from `clientsdknodeweb` so both apps will use the same version of the js-sdk.

Because we don't need `version` from the old grabthar watcher, I simplified the return of `getPayPalSmartPaymentButtonsRenderScript` and completely removed the sdk watcher.

### CODE REVIEW

**Comment by gregjopa** [comment]
PR looks good! I am seeing this flow error in CI:

```
> flow "check"
Error ------------------------------------------------------------------------- test/server/render.button.test.js:434:50
Cannot call `getButtonMiddleware` with object literal bound to the first parameter because property `sdkVersionManager`
is missing in object literal [1] but exists in `ButtonMiddlewareOptions` [2]. [prop-missing]
   test/server/render.button.test.js:434:50
                                                         v
   434|     const buttonMiddleware = getButtonMiddleware({
   435|         content: mockContent, tracking: mockTracking,
   436|         cache, logger, graphQL, getAccessToken, getMerchantID,
   437|         getPersonalizationEnabled,
   438|         getInstanceLocationInformation,
   439|         getSDKLocationInformation
   440|     });
            ^ [1]
References:
   server/components/buttons/middleware.js:55:5
    55| } : ButtonMiddlewareOptions = {}) : ExpressMiddleware {
            ^^^^^^^^^^^^^^^^^^^^^^^ [2]
Found 1 error
```

File: server/types.js [hunk 1]
```diff
@@ -2,6 +2,28 @@
 
 import { COUNTRY, LANG } from '@paypal/sdk-constants';
 
+export class SDKVersionManager {
```
Comment (wsbrunson) [inline server/types.js:5]: The type for this live in our internal github repo so we have to re-create it

File: server/components/buttons/middleware.js [hunk 2]
```diff
@@ -75,7 +87,14 @@ export function getButtonMiddleware({
...
             const clientPromise = getSmartPaymentButtonsClientScript({ debug, logBuffer, cache, useLocal, locationInformation });
-            const renderPromise = getPayPalSmartPaymentButtonsRenderScript({ logBuffer, cache, useLocal, locationInformation, sdkLocationInformation });
+            const renderPromise = getPayPalSmartPaymentButtonsRenderScript({
+                sdkCDNRegistry: sdkLocationInformation?.sdkCDNRegistry,
+                cache,
+                logBuffer,
+                useLocal,
+                sdkVersionManager
+            });
+            const sdkVersion = sdkVersionManager.getLiveVersion()
```
Comment (gregjopa) [inline server/components/buttons/middleware.js:97]:
```suggestion
            const sdkVersion = sdkVersionManager.getLiveVersion();
```

File: server/components/buttons/script.js [hunk 3]
```diff
@@ -33,42 +31,48 @@ export async function getLocalSmartPaymentButtonRenderScript() : Promise<?SmartP
...
+    const moduleDetails = await sdkVersionManager.getOrInstallSDK({
+        cdnRegistry:  sdkCDNRegistry || '',
+        childModules: [ CHECKOUT_COMPONENTS_MODULE ],
+        flat:         true,
+        dependencies: true,
+        logger:       logBuffer,
+        cache
+    })
+
+    return importDependency({
```
Comment (gregjopa) [inline server/components/buttons/script.js:71]: Awesome improvement with using `importDependency()`! 🎉

File: server/components/buttons/middleware.js (hunk 2 above)
Comment (wsbrunson) [reply server/components/buttons/middleware.js:97]: i wonder why my auto-formattor isn't working any more

**Review by gregjopa — APPROVED** [review]
💯

**Pull request merged**